core/src/lib.rs

```diff
@@ -102,3 +103,42 @@ impl<F: Fn(Pin<Box<dyn Future<Output = ()> + Send>>)> Executor for F {
         self(f)
     }
 }
+
+impl Executor for ThreadPool {
+    fn exec(&self, future: Pin<Box<dyn Future<Output = ()> + Send>>) {
+        self.spawn_ok(future)
+    }
+}
+
+#[cfg(feature = "tokio")]
+#[derive(Clone, Copy, Debug, Default)]
+pub enum TokioExecutor<'a> {
+    #[default]
+    Empty,
+    Given(&'a tokio::runtime::Runtime),
```

Comment:
I would be in favor of not supporting providing a runtime to start with unless someone comes and asks for it!

Always spawning into the global runtime should be good enough for now.

Reply:
Alright